fdbclient/BlobCipher.cpp

```diff
@@ -2362,15 +2590,29 @@ void testConfigurableEncryptionNoAuthMode(const int minDomainId) {
 		memcpy((void*)encryptedBuf.begin(), &temp[0], bufLen);
 		int tIdx = deterministicRandom()->randomInt(0, bufLen - 1);
 		temp[tIdx] += 1;
-		DecryptBlobCipherAes256Ctr decryptor(
-		    tCipherKeyKey, Reference<BlobCipherKey>(), &iv[0], BlobCipherMetrics::TEST);
+		DecryptBlobCipherAes256Ctr decryptor(tCipherKey, Reference<BlobCipherKey>(), &iv[0], BlobCipherMetrics::TEST);
 		decryptedBuf = decryptor.decrypt(&temp[0], bufLen, headerRef, arena);
 		ASSERT_NE(memcmp(decryptedBuf.begin(), &orgData[0], bufLen), 0);
+		TraceEvent("TestConfigurableEncryptionNoAuthPayloadCorruptionDone");
 	} catch (Error& e) {
 		// No authToken, hence, no corruption detection supported
 		ASSERT(false);
 	}
 
+	// induce baseCipher corruption
+	try {
+		Reference<BlobCipherKey> corruptedTextCipher = corruptCipherKey(tCipherKey);
+		encryptedBuf = encryptor.encrypt(&orgData[0], bufLen, &headerRef, arena);
+		DecryptBlobCipherAes256Ctr decryptor(
+		    corruptedTextCipher, Reference<BlobCipherKey>(), &iv[0], BlobCipherMetrics::TEST);
+		decryptedBuf = decryptor.decrypt(encryptedBuf.begin(), bufLen, headerRef, arena);
```

Comment:
Also as a side thought, should we maybe create a single workload that runs all the encryption unit tests, that way we can run this `x` times for each encryption change. Otherwise we might missing running a certain workload missing rare issues that can cause CI to block

Reply:
Good catch! Seems like copy-paste error, fix it now (plus audit all other instances)

> should we maybe create a single workload that runs all the encryption unit tests
Yeah, felt the same, we have combination of unittests and simulation tests, so having a single `toml` with different workloads might help. However, it needs to done with care as many tests are very `assertive` in nature i.e. given BlobCipherKeyCache is process-singleton and most of the test inserts a known set of keys and check for specific things. 

I think running 100/500K runs with encryption enabled should help filter all regressions (and that is what I did with this PR)

Reply:
Yeah my thought is more for the unit tests. Since our unit tests are spread across multiple toml files I think it would be worth it to consilidate to a single one so it's easier to run. I think for non-unit tests we can rely on just having encryption enabled. But that can probably be done as a separate PR if needed

Reply:
We can do it, but then many unit-test uses singleton BlobCipherCache semantics, however, we should be able to combine REST unittests into a single one and _audit/add_ BlobCipher unit test OR just add the `/blobCipher` one which covers almost all aspects.

Do you mind opening a PR to track the effort, it might be useful to have it for sure. 
Thanks